### ISSUE TITLE
SubCollections of SubCollections documentation lacking

### ISSUE THREAD

**Issue by joantune** [wovalle/fireorm#128]
Hi!

First of all - thanks for this Library -
(**Feel free to ignore - What I like, what I'm wondering about this library**) I find it very helpful to create a better ORM experience - I didn't try to do Firebase on its own - but I did come from the world of ORMs - and I like the fact that things are statically typed - without the need for boilerplate 'glue' although in the back of my mind I keep wondering if some custom made repositories and some `Object.assign` couldn't be helpful - but overall I think I would be re-doing something like fireorm - for which I prefer to use and contribute to in any way I can.

**Actual Issue**:

From my limited tests - you can have nested Subcollections with fireorm - e.g.:

```
DocA
     | DocB
             |DocC
```  

However - apparently - through the tests I saw and my implementation - I found out that you need to annotate `DocB` with `@Collection` - a paragraph on the Subcollection would be nice (I can do a PR eventually [when I'm done with this project]) LMK

**Comment by wovalle** [comment]
Hey! Thanks!

Until recently I didn't know that one subcollection could contain more subcollections that's why is not supported. 

Would you mind describing how the API would look like before doing a PR? Maybe we can discuss the implementation details before hands :)

**Comment by Mistic92** [comment]
I just found out that FIreOrm is not supporting this :( I have nested subcollections too.

**Comment by flasd** [comment]
So, you can actually have up to 25 nested collections in Firestore (coll -> doc -> coll -> doc -> coll ...) How should I go about nesting collections?

**Comment by joantune** [comment]
Sorry for the late reply - it has been crazy and hard to find time to contribute.

So - these are some code examples of how I'm doing nested collections @flasd @Mistic92  @wovalle :

```
@Collection()
export class MerchantPartner implements IEntity {
  id!: string;
[...]
 @SubCollection(Vendor)
  vendors!: ISubCollection<Vendor>;
  @SubCollection(Order)
  shopifyOrders!: ISubCollection<Order>;
[...]
}
```
Then the Vendor that has no subcollections - has no annotations - but the Order on the other hand looks like this:

```
@Collection()
export class Order implements IEntity, IShopifyEntity<Shopify.IOrder> {

[...]
  @SubCollection(LineItem)
  lineItems!: ISubCollection<LineItem>;
[...]
}
```

Hope this helps

**Comment by joantune** [comment]
So

in a scenario:
A and B require `@Collection()` annotation
C and D don't require the `@Collection()` annotation
A -> B ->C
A-> D

**Comment by wovalle** [comment]
I think that this problem is deeper than lack of documentation of how to use subcollections.

When fireorm was created I did it with the following assumptions:
- Subcollections are different than collections
- A collection can only contain one level of subcollections 

But I've learned so far that:
- Subcollections are just collections with a parent collection
- Because of the above, subcollections can also contain other subcollections

The initial abstraction around subcollections is too constrained as of today. Fixing it to start treating subcollections like normal collections (with a parent) changes entirely how fireorm does metadata lookup so I need sometime to plan it properly. I'll try to create a ticket explaining what's the problem and a possible solution involving rethinking how fireorm works with subcollections.
